### PR TITLE
Resolving sonar code smells for gNMI - 14.0.x 

### PR DESCRIPTION
**Pull request by PeterSuna**
Fix sonar code smells reported with introducing gNMI. All code smells issue should be finished except:
 - Remove usage of generic wildcard type. (We want to use NormalizeNode<?, ?> with wildcard, because yangtools is build to use NormalizedNode this way)
 - Take the required action to fix the issue indicated by this "FIXME" comment. (This would be fixed in different PR: https://github.com/PANTHEONtech/lighty/pull/758)

 https://sonarcloud.io/project/issues?directories=lighty-applications%2Flighty-rcgnmi-app-aggregator%2Flighty-rcgnmi-app-module%2Clighty-modules%2Flighty-gnmi%2Flighty-gnmi-commons%2Fsrc%2Fmain%2Fjava%2Fio%2Flighty%2Fmodules%2Fgnmi%2Fcommons%2Futil%2Clighty-modules%2Flighty-gnmi%2Flighty-gnmi-sb&id=PeterSuna_lighty-core&resolved=false&severities=CRITICAL%2CMAJOR&types=CODE_SMELL

### CODE REVIEW

**Review by marekzatko — APPROVED** [review]

File: lighty-modules/lighty-gnmi/lighty-gnmi-connector/src/main/java/io/lighty/modules/gnmi/connector/gnmi/wrappers/InstanceIdentifierNodeStreamWriter.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 
     private final StringBuilder urlBuilder;
     private static final Logger LOG = LoggerFactory.getLogger(InstanceIdentifierNodeStreamWriter.class);
+    private static final String NOT_IMPLEMENTED = "Method not implemented";
 
     public InstanceIdentifierNodeStreamWriter(String prefix) {
```
Comment (vainyksi) [inline lighty-modules/lighty-gnmi/lighty-gnmi-connector/src/main/java/io/lighty/modules/gnmi/connector/gnmi/wrappers/InstanceIdentifierNodeStreamWriter.java:27]: TODO: remove unused class

File: lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/mountpoint/broker/GnmiDataBroker.java [hunk 2]
```diff
@@ -54,7 +59,7 @@ public DOMDataTreeWriteTransaction newWriteOnlyTransaction() {
 
             @Override
             public void close() {
-
+                LOG.warn(NOT_IMPLEMENTED);
```
Comment (vainyksi) [inline lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/mountpoint/broker/GnmiDataBroker.java:62]: this is not exactly correct, no implementation is needed
change to `debug` level and extend message

File: lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/mountpoint/transactions/ReadOnlyTx.java [hunk 3]
```diff
@@ -33,7 +33,7 @@ public ReadOnlyTx(final GnmiGet getProvider) {
 
     @Override
     public void close() {
-
+        LOG.info("Method not implemented!");
```
Comment (vainyksi) [inline lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/mountpoint/transactions/ReadOnlyTx.java:36]: this is not exactly correct, no implementation is needed
change to `debug` level and extend message

**Review by vainyksi — CHANGES_REQUESTED** [review]

**Review by vainyksi — APPROVED** [review]

File: lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/mountpoint/transactions/ReadOnlyTx.java (hunk 3 above)
Comment (PeterSuna) [reply lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/mountpoint/transactions/ReadOnlyTx.java]: done

File: lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/mountpoint/broker/GnmiDataBroker.java (hunk 2 above)
Comment (PeterSuna) [reply lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/mountpoint/broker/GnmiDataBroker.java]: done

File: lighty-modules/lighty-gnmi/lighty-gnmi-connector/src/main/java/io/lighty/modules/gnmi/connector/gnmi/wrappers/InstanceIdentifierNodeStreamWriter.java (hunk 1 above)
Comment (PeterSuna) [reply lighty-modules/lighty-gnmi/lighty-gnmi-connector/src/main/java/io/lighty/modules/gnmi/connector/gnmi/wrappers/InstanceIdentifierNodeStreamWriter.java]: done

File: lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/mountpoint/codecs/GetResponseToNormalizedNodeCodec.java [hunk 4]
```diff
@@ -74,6 +74,7 @@ because the AugmentationNode has no QName so later post processing (for example
                 } else {
                     return Optional.of(codecResult);
                 }
+                return Optional.empty();
```
Comment (vainyksi) [inline lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/mountpoint/codecs/GetResponseToNormalizedNodeCodec.java:77]: this was already removed in #750

**Review by vainyksi — CHANGES_REQUESTED** [review]

**Review by vainyksi — APPROVED** [review]

**Pull request merged**